agentic/src/nodes/base.ts

```diff
@@ -257,12 +275,13 @@ export abstract class BaseNode extends KaiWorkflowEventEmitter {
     if (!this.tools || this.tools.length < 1 || !enableTools) {
       return response;
     }
+    const filteredTools = this.getToolsMatchingSelectors(toolsSelectors);
```

Comment:
we want each sub-agent to have access to only a specific set of tools. more tools != better work